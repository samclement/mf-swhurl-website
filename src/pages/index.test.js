/* eslint-env jest */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import Index from './index.js'
// Nextjs routing support
import Router from 'next/router'
const mockedRouter = { push: () => {}, prefetch: () => {} }
Router.router = mockedRouter

describe('With Snapshot Testing', () => {
  it('Index page shows "mf.swhurl.com" H1', () => {
    const { asFragment } = render(<Index />)
    expect(asFragment()).toMatchSnapshot()
  })
})
