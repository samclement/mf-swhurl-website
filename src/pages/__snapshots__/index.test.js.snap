// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Index page shows "mf.swhurl.com" H1 1`] = `
<DocumentFragment>
  .c3 {
  color: black;
}

.c2 {
  color: black;
  padding-right: 5px;
}

.c2:hover {
  color: #f00;
}

.c2:visited {
  color: black;
}

.c1 {
  font-weight: bold;
  text-transform: capitalize;
  padding-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<div>
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        mens
      </div>
      <a
        class="c2"
        href="/mens/shop/accessories"
      >
        Accessories
      </a>
      <a
        class="c2"
        href="/mens/shop/shoes"
      >
        Shoes
      </a>
      <a
        class="c2"
        href="/mens/shop/clothing"
      >
        Clothing
      </a>
      <a
        class="c2"
        href="/mens/shop/bags"
      >
        Bags
      </a>
      <a
        class="c2"
        href="/mens/shop/homeware"
      >
        Homeware
      </a>
    </nav>
    <nav
      class="c0"
    >
      <div
        class="c1"
      >
        womens
      </div>
      <a
        class="c2"
        href="/womens/shop/accessories"
      >
        Accessories
      </a>
      <a
        class="c2"
        href="/womens/shop/bags"
      >
        Bags
      </a>
      <a
        class="c2"
        href="/womens/shop/clothing"
      >
        Clothing
      </a>
      <a
        class="c2"
        href="/womens/shop/shoes"
      >
        Shoes
      </a>
      <a
        class="c2"
        href="/womens/shop/homeware"
      >
        Homeware
      </a>
    </nav>
    <h1
      class="c3"
    >
      mf.swhurl.com
    </h1>
    <p>
      Example next.js app, including:
    </p>
    <ul>
      <li>
        nextjs (server-side rendering single-page application)
      </li>
      <li>
        styled-compoenents
      </li>
      <li>
        linting with eslint
      </li>
      <li>
        jest snapshot testing
      </li>
      <li>
        prettier formatting
      </li>
      <li>
        lighthouse performance testing
      </li>
      <li>
        chrome browser testing (wdio)
      </li>
      <li>
        greenkeeper (automatic PRs for upgrading dependencies)
      </li>
      <li>
        kustomize (k8s deployment configuration)
      </li>
    </ul>
    <p>
      The application is deployed to a k8s cluster that includes:
    </p>
    <ul>
      <li>
        cert-manager (automatic TLS certificate provisioning from LetsEncript)
      </li>
      <li>
        oauth2-proxy (authentication against github)
      </li>
      <li>
        prometheus (infrastructure and applicaiton monitoring)
      </li>
      <li>
        fluentbit / loggly integration (log aggregation)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
